Packs/Whois/Integrations/Whois/Whois.py

```diff
@@ -8297,28 +8322,34 @@ def parse_nic_contact(data):
     return handle_contacts
 
 
-def get_whois(domain, normalized=None, is_recursive=True):
-    if normalized is None:
-        normalized = []
+def get_whois(domain: str, is_recursive=True):
+
     raw_data, server_list = get_whois_raw(domain, with_server_list=True, is_recursive=is_recursive)
-    return parse_raw_whois(raw_data, normalized=normalized, never_query_handles=False,
+    return parse_raw_whois(raw_data, normalized=[], never_query_handles=False,
                            handle_server=server_list[-1])
 
 
 # Drops the mic disable-secrets-detection-end
 
 def get_domain_from_query(query):
-    # checks for largest matching suffix inside tlds dictionary
-    suffix_len = max([len(suffix) for suffix in tlds if query.endswith('.{}'.format(suffix))] or [0])
-    # if suffix(TLD) was found increase the length by one in order to add the dot before it. --> .com instead of com
-    if suffix_len != 0:
-        suffix_len += 1
-    suffixless_query = query[:-suffix_len]
-    domain = query
-    # checks if query includes subdomain
-    if suffixless_query.count(".") > 0:
-        domain = query[suffixless_query.rindex(".") + 1:]
-    return domain
+
+    demisto.debug(f"Attempting to get domain from query '{query}'...")
+
+    try:
+        # checks for largest matching suffix inside tlds dictionary
+        suffix_len = max([len(suffix) for suffix in tlds if query.endswith('.{}'.format(suffix))] or [0])
+        # if suffix(TLD) was found increase the length by one in order to add the dot before it. --> .com instead of com
+        if suffix_len != 0:
+            suffix_len += 1
+        suffixless_query = query[:-suffix_len]
+        domain = query
+        # checks if query includes subdomain
+        if suffixless_query.count(".") > 0:
+            domain = query[suffixless_query.rindex(".") + 1:]
```

Comment:
Does this work for both `amazon.com` and `amazon.co.uk`?

Reply:
Yep, there's a UT specifically for that:

```python
@pytest.mark.parametrize(
    'query,expected',
    [("app.paloaltonetwork.com", "paloaltonetwork.com"),
     ("test.this.google.co.il", "google.co.il"),
     ("app.XSOAR.test", "app.XSOAR.test")]
)
def test_get_domain_from_query(query, expected):
    assert get_domain_from_query(query) == expected
```